diff --git a/tools/winebuild/import.c b/tools/winebuild/import.c
index 5f62210f486..89fc5267a23 100644
--- a/tools/winebuild/import.c
+++ b/tools/winebuild/import.c
@@ -1304,29 +1304,9 @@ void output_stubs( DLLSPEC *spec )

             output( "\tsubl $12,%%esp\n" );
             output_cfi( ".cfi_adjust_cfa_offset 12" );
-            if (UsePIC)
-            {
-                output( "\tcall %s\n", asm_name("__wine_spec_get_pc_thunk_eax") );
-                output( "1:" );
-                needs_get_pc_thunk = 1;
-                if (exp_name)
-                {
-                    output( "\tleal .L%s_string-1b(%%eax),%%ecx\n", name );
-                    output( "\tmovl %%ecx,4(%%esp)\n" );
-                }
-                else
-                    output( "\tmovl $%d,4(%%esp)\n", odp->ordinal );
-                output( "\tleal .L__wine_spec_file_name-1b(%%eax),%%ecx\n" );
-                output( "\tmovl %%ecx,(%%esp)\n" );
-            }
-            else
-            {
-                if (exp_name)
-                    output( "\tmovl $.L%s_string,4(%%esp)\n", name );
-                else
-                    output( "\tmovl $%d,4(%%esp)\n", odp->ordinal );
-                output( "\tmovl $.L__wine_spec_file_name,(%%esp)\n" );
-            }
+            output( "\t.byte 0xb8\n" );                               /* mov eax, SYSCALL */
+            output( "\t.long %d\n", i );
+            output( "\t.byte 0x64,0xff,0x15,0xc0,0x00,0x00,0x00\n" ); /* call dword ptr fs:[0C0h] */
             output( "\tcall %s\n", asm_name("__wine_spec_unimplemented_stub") );
             break;
         case CPU_x86_64:
