diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index c69d644ebfe..53b2441bc3a 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -102,7 +102,8 @@ AddReg=\
     Timezones,\
     VersionInfo,\
     LicenseInformation,\
-    LutrisClient
+    LutrisClient, \
+    NVIDIANGX

 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -131,7 +132,8 @@ AddReg=\
     Timezones,\
     VersionInfo.ntamd64,\
     LicenseInformation,\
-    LutrisClient.ntamd64
+    LutrisClient.ntamd64, \
+    NVIDIANGX

 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -175,7 +177,8 @@ AddReg=\
     Tapi,\
     VersionInfo.ntamd64,\
     LicenseInformation,\
-    LutrisClient.ntamd64
+    LutrisClient.ntamd64, \
+    NVIDIANGX

 [Wow64Install.ntarm64]
 WineFakeDlls=FakeDllsWin32
@@ -6091,6 +6094,7 @@ HKCU,Software\Wine\DllOverrides,"ucrtbase",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcomp140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"atiadlxx",,"disabled"
+HKCU,Software\Wine\DllOverrides,"nvcuda",0x2,"disabled"

 [LutrisClient.ntamd64]
 HKCU,Software\Wine\AppDefaults\LauncherPatcher.exe,"Version",,"win7"
@@ -6147,3 +6151,7 @@ HKCU,Software\Wine\DllOverrides,"ucrtbase",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcomp140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"atiadlxx",,"disabled"
+HKCU,Software\Wine\DllOverrides,"nvcuda",0x2,"disabled"
+
+[NVIDIANGX]
+HKLM,Software\NVIDIA Corporation\Global\NGXCore,"FullPath",,"C:\Windows\System32"
