diff --git a/dlls/imm32/ime.c b/dlls/imm32/ime.c
index 38a2a070117..424b286d0b5 100644
--- a/dlls/imm32/ime.c
+++ b/dlls/imm32/ime.c
@@ -701,6 +701,13 @@ BOOL WINAPI NotifyIME( HIMC himc, DWORD action, DWORD index, DWORD value )
         }
         case CPS_CANCEL:
             input_context_set_comp_str( ctx, NULL, 0 );
+            const char *sgi = getenv("SteamGameId");
+
+            if ((!sgi) | (sgi && !strcmp(sgi, "552990"))) {
+                ImmSetOpenStatus( himc, FALSE );
+		 break;
+            };
+
             if ((msg = ime_set_composition_status( himc, FALSE ))) ime_send_message( himc, msg, 0, 0 );
             NtUserNotifyIMEStatus( ctx->hWnd, FALSE );
             break;
