diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index e7c10e17f38..ecaeb33f3fe 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -2865,3 +3105,5 @@ HKCU,Software\Wine\AppDefaults\Maine-Win64-Shipping.exe\DllOverrides,"SpeechSynt
 HKCU,Software\Wine\AppDefaults\rayne1.exe\DllOverrides,"d3d8",,"native"
 HKCU,Software\Wine\AppDefaults\rayne2.exe\DllOverrides,"d3d8",,"native"
 HKCU,Software\Wine\AppDefaults\RDR2.exe\DllOverrides,"vulkan-1",,"native"
+HKCU,Software\Wine\AppDefaults\RiotClientUxRender.exe\DllOverrides,"winegstreamer",,""
+HKCU,Software\Wine\AppDefaults\LeagueClientUxRender.exe\DllOverrides,"winegstreamer",,""
